docs/sources/k6/next/using-k6/secret-source/_index.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Secret Source
+description: 'Secret source are used so k6 can get and use secrets in a secure way'
+weight: 14
+---
+
+# Secret source
+
+Secret sources are a way for k6 to acquire secrets to be used with k6.
+
+The secrets are made available through the `[k6/secrets`]() JS API and will be redacted from any logs.
+
+## Configure secret sources
+
+Currently the only way to configured secret sources is through the `--secret-source` cli flag. Multiple secret sources are configurable at the same time
+
+## Secret sources
+
+Current build-in secret sources are limited and are mostly meant for local testing:
+
+- [`file`](https://grafana.com/docs/k6/<K6_VERSION>/using-k6/secret-source/file) reads secrets from a key=value file.
+- [`mock`](https://grafana.com/docs/k6/<K6_VERSION>/using-k6/secret-source/mock) reads secrets from the cli flag.
+
+## Secret source extensions
+
+You can implement a secret source as an [extension](https://grafana.com/docs/k6/<K6_VERSION>/extensions/) for k6.
+
+## Example script
+
+{{< code >}}
+
+<!-- md-k6:skip -->
+
+```javascript
+import secrets from 'k6/secrets';
+
+export default async () => {
+  const my_secret = await secrets.get('cool'); // get secret from a source with the provided identifier
+  console.log(my_secret);
+  await secrets.get('else'); // get secret from a source with the provided identifier
+  console.log(my_secret);
```

Comment:
It would be good to have a more real code. I expect people is interested into _using_ the secret more than logging it.

A classic HTTP request with the API token passed in Authz header might be a good example.